Update changelogs for 1.5.4

jacek-casper opened the pull request:
The changelog here is pretty detailed, let me know if something can be removed, I opted on the side of keeping things in. Also added PR numbers, but I can remove them if we don't want them.

jacek-casper committed:
Update changelogs for 1.5.4

MParlikar reviewed (commented):
Is there a changelog for the chainspec? The gas table + block times are also changing.

jacek-casper commented:
@MParlikar there is no changelog for the chainspec AFAIK, but we could add it under the execution engine, because that's where the opcode costs are used (and the PR actually changed the default costs there too: https://github.com/casper-network/casper-node/pull/4387/files)

alsrdn commented on execution_engine/CHANGELOG.md:
```diff
@@ -21,12 +21,15 @@ All notable changes to this project will be documented in this file.  The format
 * Add chainspec option `core.refund_handling` that specifies how payment refunds are handled.
 * Add chainspec option `core.fee_handling` that specifes how transaction fees are handled.
 * Add chainspec option `core.administrators` that, if set, contains list of administrator accounts. This option makes sense only for private chains.
+* Add initial support for private chain (#3106). 
```
It would be good here to unwrap the new chainspec parameters added with the private chain changes since they are user facing.

jacek-casper replied:
I've gone through them @alsrdn and I noticed that the list right above includes all of the new private chains parameters that I can see (`core.administrators`, `core.fee_handling`, etc)
I'm going to remove this line from the changelog

alsrdn commented on execution_engine/CHANGELOG.md:
```diff
@@ -21,12 +21,15 @@ All notable changes to this project will be documented in this file.  The format
 
-
+### Fixed
+* Fix an issue where payment errors penalty could've been paid to block proposer regardless of fee_handling/refund_handling option (#4258).
```
As I understand it this issue has never been part of a release since it was introduced due to merging; so we may not need to add it in the readme. But not sure about this, maybe @mpapierski can comment.

alsrdn commented on node/CHANGELOG.md:
```diff
@@ -13,14 +13,33 @@ All notable changes to this project will be documented in this file.  The format
 * New environment variable `CL_EVENT_QUEUE_DUMP_THRESHOLD` to enable dumping of queue event counts to log when a certain threshold is exceeded.
+* Add initial support for private chain (#3106). 
+* Add support for CA signed client certificates for private chain (#3310).
+* Add a finer grained latching mechanism for block_synchronization (#4042).
```
This was part of 1.5.3 and is described below with:
`Remove the `block_synchronizer.stall_limit` node config parameter since it is no longer needed` and `Change the limit of the `core_config.simultaneous_peer_requests` chainspec `

alsrdn commented on node/CHANGELOG.md:
```diff
@@ -13,14 +13,33 @@ All notable changes to this project will be documented in this file.  The format
+* Add initial support for private chain (#3106). 
+* Add support for CA signed client certificates for private chain (#3310).
+* Add a finer grained latching mechanism for block_synchronization (#4042).
+* Add a NoSync option, which allows a node to opt out of historical sync via config setting (#4257).
```
I think this should be moved to the `Changed` section and we should mention that the `sync_to_genesis` parameter was replaced with 3 other options:

```
# Historical sync behavior for this node. Options are:
#  'genesis'  (node will attempt to acquire all block data back to genesis)
#  'ttl'      (node will attempt to acquire all block data to comply with time to live enforcement)
#  'nosync'   (node will only acquire blocks moving forward)
```

alsrdn commented on node/CHANGELOG.md:
```diff
@@ -13,14 +13,33 @@ All notable changes to this project will be documented in this file.  The format
+* Add a finer grained latching mechanism for block_synchronization (#4042).
+* Add a NoSync option, which allows a node to opt out of historical sync via config setting (#4257).
+* Add more detail in the log message for too many events in the queue (#4295).
+* Add more logging in BlockValidator (#4299).
```
I don't think it's worth mentioning these in the changelog. These were only additional logs to help with debug, but I guess it doesn't hurt to have them here either.

alsrdn commented on node/CHANGELOG.md:
```diff
@@ -13,14 +13,33 @@ All notable changes to this project will be documented in this file.  The format
 
 ### Fixed
 * Now possible to build outside a git repository context (e.g. from a source tarball). In such cases, the node's build version (as reported vie status endpoints) will not contain a trailing git short hash.
+* Purge block synchronizer every time we transition out of CatchUp (#4058).
+* Don't return a fatal error when trying to deactivate an era that was already deactivated (#4075).
```
These were internal fixes that were part of 1.5.3. Should be removed from here.

alsrdn commented on node/CHANGELOG.md:
```diff
@@ -13,14 +13,33 @@ All notable changes to this project will be documented in this file.  The format
+* Purge block synchronizer every time we transition out of CatchUp (#4058).
+* Don't return a fatal error when trying to deactivate an era that was already deactivated (#4075).
+* Fix historical sync stall after restart in era 0 on pre-1.5 chain (#4129).
+* Remove error for timestamp chronology to resolve failing sync (#4195).
```
This 2 fixes go together. #4195 was a fix for an issue introduced in #4129. I would add a single line to the Readme describing the issue and what was fixed without citing the PRs here.

alsrdn commented on node/CHANGELOG.md:
```diff
@@ -13,14 +13,33 @@ All notable changes to this project will be documented in this file.  The format
+* Don't return a fatal error when trying to deactivate an era that was already deactivated (#4075).
+* Fix historical sync stall after restart in era 0 on pre-1.5 chain (#4129).
+* Remove error for timestamp chronology to resolve failing sync (#4195).
+* Fix block synchronizer request storm by tweaking the latching (#4205).
```
This was also part of 1.5.3 and it is an improvement to the `Add a finer grained latching mechanism for block_synchronization PR`. Should be removed from this section.

alsrdn commented on node/CHANGELOG.md:
```diff
@@ -13,14 +13,33 @@ All notable changes to this project will be documented in this file.  The format
 
 ### Changed
 * The `state_identifier` parameter of the `query_global_state` JSON-RPC method is now optional. If no `state_identifier` is specified, the highest complete block known to the node will be used to fulfill the request.
+* Make the state_identifier parameter of the query_global_state JSON-RPC method optional (#4086).
```
This is already documented in the line just before this one.

alsrdn commented on node/CHANGELOG.md:
```diff
@@ -13,14 +13,33 @@ All notable changes to this project will be documented in this file.  The format
 ### Changed
 * The `state_identifier` parameter of the `query_global_state` JSON-RPC method is now optional. If no `state_identifier` is specified, the highest complete block known to the node will be used to fulfill the request.
+* Make the state_identifier parameter of the query_global_state JSON-RPC method optional (#4086).
+* Downgrade the 'highest_era_weights does not contain signing_public_key' log message to DEBUG (#4143).
```
Like in the case of the other debug logging message changes, not sure it is a worthwhile mention but don't have a hard preference here.

alsrdn commented:
> @MParlikar there is no changelog for the chainspec AFAIK, but we could add it under the execution engine, because that's where the opcode costs are used (and the PR actually changed the default costs there too: https://github.com/casper-network/casper-node/pull/4387/files)

I think mentioning the default cost changes in the REDME of the EE is a good idea.
We also provide some chainspec examples in the node but not sure if we should actually include this in the node README.

I think it may be best to add a changelog to this repo when the cost changes get merged: https://github.com/casper-network/casper-protocol-release
@MParlikar @jacek-casper WDYT?

jacek-casper committed:
Address PR comments

jacek-casper committed:
Mention lowering default gas opcode costs

jacek-casper commented:
@alsrdn thanks for the review, I addressed all your comments,
I don't mind adding a changelog in there if that's the best place for production chainspec changes, namely these three things probably need to be mentioned: 
- lower the production gas cost for opcodes ...
- lower the minimum block time to 16 seconds
- lower the upper limit of total gas of all deploys in a block to 4,000,000,000,00

jacek-casper committed:
Fix bad PR number

alsrdn reviewed (approved):
LGTM

Fraser999 commented on execution_engine/CHANGELOG.md:
```diff
@@ -22,11 +22,13 @@ All notable changes to this project will be documented in this file.  The format
 ### Changed
 * Default value for `max_stack_height` is increased to 500.
 * `current stack height` is written to `stderr` in case `Trap(Unreachable)` error is encountered during Wasm execution.
+* Tweaked upgrade logic transforming withdraw purses to early exit if possible (#4119).
```
```suggestion
* Tweak upgrade logic transforming withdraw purses to early exit if possible (#4119).
```

Fraser999 commented on execution_engine/CHANGELOG.md:
```diff
@@ -22,11 +22,13 @@ All notable changes to this project will be documented in this file.  The format
+* Lower the default gas costs of opcodes (#4387).
+  - Set the cost for branching opcodes to 35,000 (`br`, `br_if`, `br_table`).
+  - Set the cost for call opcodes to 68,000 (`call`, `call_indirect`).
 
```
```suggestion



```

Fraser999 commented on node/CHANGELOG.md:
```diff
@@ -13,15 +13,28 @@ All notable changes to this project will be documented in this file.  The format
 
 ## Unreleased
 
+### Security
```
Could the order of the sections be made to match the commented example lines above please?

Fraser999 commented on node/CHANGELOG.md:
```diff
@@ -13,15 +13,28 @@ All notable changes to this project will be documented in this file.  The format
+    - `genesis` - node will attempt to acquire all block data back to genesis
+    - `ttl` - node will attempt to acquire all block data to comply with time to live enforcement
+    - `nosync` - node will only acquire blocks moving forward
 
```
```suggestion


```

Fraser999 commented on types/CHANGELOG.md:
```diff
@@ -16,7 +16,11 @@ All notable changes to this project will be documented in this file.  The format
 ### Security
 * Update `ed25519-dalek` to version 2.0.0 as mitigation for [RUSTSEC-2022-0093](https://rustsec.org/advisories/RUSTSEC-2022-0093)
 
+### Added
```
Same comment about the section ordering.

Fraser999 commented on types/CHANGELOG.md:
```diff
@@ -16,7 +16,11 @@ All notable changes to this project will be documented in this file.  The format
 
+### Changed
+* Update `k256` to version 0.13.1 (#4191).
 
```
```suggestion



```

Fraser999 commented on node/CHANGELOG.md:
```diff
@@ -13,15 +13,28 @@ All notable changes to this project will be documented in this file.  The format
-
+* `state_get_account_info` RPC handler can now handle an `AccountIdentifier` as a parameter (#4218).
+* Replace the `sync_to_genesis` node config field with `sync_handling` (#4257).
+  * The new field `sync_handling` field accepts three values:
```
```suggestion
  * The new `sync_handling` field accepts three values:
```

Fraser999 reviewed (commented):
I'd prefer if the PR numbers were removed.  Main reason is down to not being able to apply them consistently, since some PRs will be made to the private repo and can't be found publicly.

If we do end up keeping them, I'd suggest adding actual links, so they become clickable.

jacek-casper committed:
Address PR comments

jacek-casper committed:
Remove PR numbers

jacek-casper commented:
@Fraser999 I addressed all the comments, let me know if there's anything else

jacek-casper committed:
Mention enable_contract_version

jacek-casper commented:
@mpapierski I've added a mention of `enable_contract_version`, can you have a look and approve?

Fraser999 reviewed (approved):
LGTM - thanks for the updates!

jacek-casper commented:
bors r+

cspramit commented:
The final change logs - https://github.com/casper-network/casper-node/blob/dev/execution_engine/CHANGELOG.md#changed     as a sampling has errors it is showing changed for stuff which was done in past release such as max_stack height 500

jacek-casper commented:
@cspramit I didn't remove anything from the Unreleased blocks, because I've assumed they were left there intentionally, I can open another PR removing things that don't appear in the 1.5.3..dev diff
